config.py

```diff
@@ -742,157 +754,183 @@ def check_yaml(conf_path):
-                                "and will be removed in a future version. "
-                                "Please use conf.d")
-                    check_config = {'instances':[dict((key, agentConfig[key]) for key in agentConfig if key in NAGIOS_OLD_CONF_KEYS)]}
-                else:
-                    continue
             else:
-                log.debug("No configuration file for %s" % check_name)
-                continue
+                break
 
-        # If we are here, there is a valid matching configuration file.
-        # Let's try to import the check
+    if not check_class:
+        log.error('No check class (inheriting from AgentCheck) found in %s.py' % check_name)
+        return False, None, {}
+    return True, check_class, {}
+
+def _initialize_check(check_config, check_name, check_class, agentConfig):
+    init_config = check_config.get('init_config') or {}
+    instances = check_config['instances']
+    try:
         try:
-            check_module = imp.load_source('checksd_%s' % check_name, check)
-        except Exception, e:
-            traceback_message = traceback.format_exc()
-            # There is a configuration file for that check but the module can't be imported
-            init_failed_checks[check_name] = {'error':e, 'traceback':traceback_message}
-            log.exception('Unable to import check module %s.py from checks.d' % check_name)
-            continue
+            check = check_class(check_name, init_config=init_config,
+                            agentConfig=agentConfig, instances=instances)
+        except TypeError, e:
+            # Backwards compatibility for checks which don't support the
+            # instances argument in the constructor.
+            check = check_class(check_name, init_config=init_config,
+                            agentConfig=agentConfig)
+            check.instances = instances
+    except Exception, e:
+        log.exception('Unable to initialize check %s' % check_name)
+        traceback_message = traceback.format_exc()
+        return {}, {check_name: {'error':e, 'traceback':traceback_message}}
+    else:
+        return {check_name: check}, {}
 
-        # We make sure that there is an AgentCheck class defined
-        check_class = None
-        classes = inspect.getmembers(check_module, inspect.isclass)
-        for _, clsmember in classes:
-            if clsmember == AgentCheck:
-                continue
-            if issubclass(clsmember, AgentCheck):
-                check_class = clsmember
-                if AgentCheck in clsmember.__bases__:
-                    continue
-                else:
-                    break
+def _update_python_path(check_config):
+    # Add custom pythonpath(s) if available
+    if 'pythonpath' in check_config:
+        pythonpath = check_config['pythonpath']
+        if not isinstance(pythonpath, list):
+            pythonpath = [pythonpath]
```

Comment:
Does it make sense to do some stricter checks here? For example `[os.path.exists(path) for path in pythonpath]`? You probably know better whether that case should fail the initialization, print a warning, or do nothing at all